crates/holochain_cascade/tests/tests/get_activity.rs

```diff
@@ -61,18 +62,142 @@ async fn get_activity() {
+        fill_db(&dht.to_db(), hash_op).await;
+    }
+    for hash_op in td.store_ops.iter().cloned() {
+        fill_db(&cache.to_db(), hash_op).await;
+    }
+
+    let warrant = {
+        let action_pair = (
+            (td.hash_ops[0].action().to_hash(), ::fixt::fixt!(Signature)),
+            (td.hash_ops[1].action().to_hash(), ::fixt::fixt!(Signature)),
+        );
+        let p = WarrantProof::ChainIntegrity(ChainIntegrityWarrant::ChainFork {
+            chain_author: td.agent.clone(),
+            action_pair,
+        });
+        // let p = WarrantProof::ChainIntegrity(ChainIntegrityWarrant::InvalidChainOp {
+        //     action_author: td.agent.clone(),
+        //     action: (td.chain_head.hash.clone(), ::fixt::fixt!(Signature)),
+        //     validation_type: ValidationType::Sys,
+        // });
+        let warrant = Warrant::new(p, AgentPubKey::from_raw_36(vec![255; 36]), Timestamp::now());
+        WarrantOp::from(SignedWarrant::new(warrant, ::fixt::fixt!(Signature)))
+    };
+
+    // Insert unvalidated warrant op
+    dht.test_write({
+        let op = DhtOp::from(warrant.clone()).into_hashed();
+        move |txn| {
+            holochain_state::mutations::insert_op(txn, &op).unwrap();
+        }
+    });
+
+    let options = holochain_p2p::actor::GetActivityOptions {
+        include_valid_activity: true,
+        include_rejected_activity: false,
+        include_full_actions: true,
+        ..Default::default()
+    };
+
+    // Network
+    let network = PassThroughNetwork::authority_for_nothing(vec![dht.to_db().clone().into()]);
+
+    // Cascade
+    let cascade = CascadeImpl::empty().with_network(network, cache.to_db());
+
+    let mut expected = AgentActivityResponse {
+        agent: td.agent.clone(),
+        valid_activity: td.valid_records.clone(),
+        rejected_activity: ChainItems::NotRequested,
+        warrants: vec![],
+        status: ChainStatus::Valid(td.chain_head.clone()),
+        highest_observed: Some(td.highest_observed.clone()),
+    };
+
+    let r1 = cascade
+        .get_agent_activity(td.agent.clone(), ChainQueryFilter::new(), options.clone())
+        .await
+        .unwrap();
+
+    assert_eq!(r1, expected);
+
+    // If the warrant is validated, it will be returned
+    dht.test_write({
+        let op = DhtOp::from(warrant.clone()).into_hashed();
+        let op_hash = op.as_hash().clone();
+        move |txn| {
+            holochain_state::mutations::set_validation_status(
+                txn,
+                &op_hash,
+                ValidationStatus::Valid,
+            )
+            .unwrap();
+            holochain_state::mutations::set_when_integrated(txn, &op_hash, Timestamp::now())
+                .unwrap();
+        }
+    });
+
+    let r2 = cascade
+        .get_agent_activity(td.agent.clone(), ChainQueryFilter::new(), options)
+        .await
+        .unwrap();
+
+    expected.warrants = vec![warrant.into_warrant()];
+
+    assert_eq!(r2, expected);
+}
```

Comment:
Nice test! Comments explaining the highlights are appreciated, they make it much easier to follow the test logic :)